### DISCUSSION TITLE
Animation question for help

### DISCUSSION THREAD

**Discussion by BonanzaBegin** [AvaloniaUI/Avalonia#13361] [Q&A] [answered]
i want to use animation to show something special,but some questions confused me ,and i have no idea with these problems 

xaml code below:
        <Border x:Name="border" Width="100" Height="100" Background="LightGreen" VerticalAlignment="Center" Margin="0,0,0,0"
                BorderBrush="Green">
            <Border.Styles>
                <Style Selector="Border:pointerover">
                    <Style.Animations>
                        <Animation Duration="0:0:2" FillMode="Forward" PlaybackDirection="Normal">
                            <KeyFrame Cue="0%">
                                <Setter Property="Background" Value="Blue"/>
                                <Setter Property="BorderThickness" Value="0"/>
                                <Setter Property="Width" Value="100"/>
                                <Setter Property="Height" Value="100"/>
                            </KeyFrame>
                            <KeyFrame Cue="100%">
                                <Setter Property="Background" Value="Red"/>
                                <Setter Property="BorderThickness" Value="10"/>
                                <Setter Property="Width" Value="200"/>
                                <Setter Property="Height" Value="200"/>
                            </KeyFrame>
                        </Animation>
                    </Style.Animations>
                </Style>
            </Border.Styles>
        </Border>

the xaml file when pointerover the animation behaviours work very well

but here is the problem  ,i wrote the animation by codebehind :

    private async void Btn_Click(object? sender, Avalonia.Interactivity.RoutedEventArgs e)
    {
        await DoAnimation();
    }

    private async Task DoAnimation()
    {
        await Task.Delay(1);
        Animation animation = new Animation();
        animation.Duration = TimeSpan.FromMilliseconds(2000);
        animation.FillMode = FillMode.Forward;
        animation.PlaybackDirection = PlaybackDirection.Normal;
        var kf1 = new KeyFrame() { Cue = new Cue(0) };
        kf1.Setters.Add(new Setter() { Property = Border.BackgroundProperty, Value = Brushes.Blue });
        kf1.Setters.Add(new Setter() { Property = Border.BorderThicknessProperty, Value = new Thickness(0) });
        kf1.Setters.Add(new Setter() { Property = Border.WidthProperty, Value = 100 });
        kf1.Setters.Add(new Setter() { Property = Border.HeightProperty, Value = 100 });

        var kf2 = new KeyFrame() { Cue = new Cue(1.0) };
        kf2.Setters.Add(new Setter() { Property = Border.BackgroundProperty, Value = Brushes.Red });
        kf2.Setters.Add(new Setter() { Property = Border.BorderThicknessProperty, Value = new Thickness(10) });
        kf2.Setters.Add(new Setter() { Property = Border.WidthProperty, Value = 300 });
        kf2.Setters.Add(new Setter() { Property = Border.HeightProperty, Value = 300 });
        animation.Children.Add(kf1);
        animation.Children.Add(kf2);
        await animation.RunAsync(this.border);

    }

if WithProperty and HeightProperty added to the KeyFrame ,when i click the button, the animation behavior didn't work ,even the border disapprear.......

who can help me  to fix the problem? thanks a lot

**Comment by BonanzaBegin** [comment]
![image](https://github.com/AvaloniaUI/Avalonia/assets/58451874/a2c7f27d-6063-4018-97b1-4c6e52a2b2be)

the animation use xaml works very well.......

**Comment by dme-compunet** [comment, marked as answer]
@BonanzaBegin 

In these lines:
```
kf1.Setters.Add(new Setter() { Property = Border.WidthProperty, Value = 100 });
kf1.Setters.Add(new Setter() { Property = Border.HeightProperty, Value = 100 });
```
you are trying to insert an `Int32` value and the `WidthProperty` and `HeightProperty` properties are of type `Double`, you need change it to something like `300d` or `300.0`

**Reply by BonanzaBegin** [reply]
yes it works...., thanks a lot ....
